x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -363,4 +368,99 @@ export function registerIndexRoutes({
+    {
+      path: '/internal/enterprise_search/indices/{indexName}/pipelines/ml_inference/_simulate',
+      validate: {
+        body: schema.object({
+          pipeline: schema.object({
+            description: schema.maybe(schema.string()),
+            processors: schema.arrayOf(schema.any()),
+          }),
+          docs: schema.arrayOf(schema.any()),
+        }),
+        params: schema.object({
+          indexName: schema.string(),
+        }),
+      },
+    },
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const { pipeline, docs } = request.body;
+      const indexName = decodeURIComponent(request.params.indexName);
+      const { client } = (await context.core).elasticsearch;
+      const defaultDescription = `ML inference pipeline for index ${indexName}`;
+
+      if (!(await indexOrAliasExists(client, indexName))) {
+        return createError({
+          errorCode: ErrorCode.INDEX_NOT_FOUND,
+          message: `Could not find index [${indexName}]`,
+          response,
+          statusCode: 404,
+        });
+      }
+
+      const simulateRequest: IngestSimulateRequest = {
+        docs,
+        pipeline: { description: defaultDescription, ...pipeline },
+      };
+
+      const simulateResult = await client.asCurrentUser.ingest.simulate(simulateRequest);
+
+      return response.ok({
+        body: simulateResult,
+        headers: { 'content-type': 'application/json' },
+      });
+    })
+  );
+
+  router.put(
+    {
+      path: '/internal/enterprise_search/indices/{indexName}/pipelines/ml_inference/{pipelineName}',
+      validate: {
+        body: schema.object({
+          description: schema.maybe(schema.string()),
+          processors: schema.arrayOf(schema.any()),
+        }),
+        params: schema.object({
+          indexName: schema.string(),
+          pipelineName: schema.string(),
+        }),
+      },
+    },
+    elasticsearchErrorHandler(log, async (context, request, response) => {
+      const pipelineBody = request.body;
+      const indexName = decodeURIComponent(request.params.indexName);
+      const pipelineName = decodeURIComponent(request.params.pipelineName);
+      const { client } = (await context.core).elasticsearch;
+      const pipelineId = `ml-inference-${pipelineName}`;
+      const defaultDescription = `ML inference pipeline for index ${indexName}`;
+
+      if (!(await indexOrAliasExists(client, indexName))) {
+        return createError({
+          errorCode: ErrorCode.INDEX_NOT_FOUND,
+          message: `Could not find index [${indexName}]`,
+          response,
+          statusCode: 404,
+        });
+      }
+
+      const updateRequest: IngestPutPipelineRequest = {
+        _meta: {
+          managed: true,
+          managed_by: 'Enterprise Search',
+        },
+        id: pipelineId,
+        description: defaultDescription,
+        ...pipelineBody,
+      };
+
+      await client.asCurrentUser.ingest.putPipeline(updateRequest);
```

Comment:
FYI - just a note here - this is great for updating the pipeline created by @demjened 's work on the [current creation flow back end](https://github.com/elastic/kibana/pull/140645) - when / if we incorporate this code into the UI flow, we'll want to use those pieces (that not only creates the pipeline, but also updates the references in the parent pipeline as well)